src/main/java/no/bibsys/aws/secrets/AWSSecretsReader.java

```diff
@@ -0,0 +1,59 @@
+package no.bibsys.aws.secrets;
+
+import com.amazonaws.regions.Region;
+import com.amazonaws.services.secretsmanager.AWSSecretsManager;
+import com.amazonaws.services.secretsmanager.AWSSecretsManagerClientBuilder;
+import com.amazonaws.services.secretsmanager.model.GetSecretValueRequest;
+import com.amazonaws.services.secretsmanager.model.GetSecretValueResult;
+import com.amazonaws.services.secretsmanager.model.InvalidRequestException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import no.bibsys.aws.tools.JsonUtils;
+
+import java.io.IOException;
+import java.util.Optional;
+
+public class AWSSecretsReader implements SecretsReader {
```

Comment:
Can we look at the use of Optional here? Is it possible to use a class that presents a coherent interface?

Reply:
Optional is used when one asks for a value that may or may not be  
Here we are requesting the value of a secret. It may  or may not be stored. 
The private method is aware that there might not be a value there, so it returns an Optional.
The public method returns a normal String to indicate that the Secret is expected to be there and it is an error if it does not exist.